subsys/usb/device_next/class/bt_hci.c

```diff
@@ -0,0 +1,491 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Bluetooth HCI USB transport layer implementation with following
+ * endpoint configuration:
+ *  - HCI commands through control endpoint (host-to-device only)
+ *  - HCI events through interrupt IN endpoint
+ *  - ACL data through one bulk IN and one bulk OUT endpoints
+ *
+ * TODO: revise transfer handling so that Bluetooth net_bufs can be used
+ * directly without intermediate copying.
+ *
+ * Limitations:
+ *  - Remote wakeup before IN transfer is not yet supported.
+ *  - H4 transport layer is not yet supported
+ */
+
+#include <zephyr/init.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <zephyr/usb/usbd.h>
+#include <zephyr/usb/usb_ch9.h>
+#include <zephyr/drivers/usb/udc.h>
+
+#include <zephyr/net/buf.h>
+
+#include <zephyr/bluetooth/buf.h>
+#include <zephyr/bluetooth/hci_raw.h>
+#include <zephyr/bluetooth/l2cap.h>
+#include <zephyr/bluetooth/hci_vs.h>
+#include <zephyr/drivers/bluetooth/hci_driver.h>
+#include <zephyr/sys/atomic.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(bt_hci, CONFIG_USBD_BT_HCI_LOG_LEVEL);
+
+#define USB_BT_HCI_SUBCLASS		0x01
+#define USB_BT_HCI_PROTOCOL		0x01
+
+#define BT_HCI_DEFAULT_BULK_EP_MPS	0	/* Get maximum supported */
+#define BT_HCI_DEFAULT_INT_EP_MPS	64
+#define BT_HCI_CLASS_ENABLED		0
+#define BT_HCI_ACL_RX_ENGAGED		1
+
+static K_FIFO_DEFINE(bt_hci_rx_queue);
+static K_FIFO_DEFINE(bt_hci_tx_queue);
+
+/*
+ * REVISE: global (bulk, interrupt, iso) specific pools would be more
+ * RAM usage efficient.
+ */
+NET_BUF_POOL_FIXED_DEFINE(bt_hci_ep_pool,
+			  3, 512,
```

Comment:
Why these specific numbers?

Reply:
Three endpoints, maximum three transfers used synchronously, 512 is the maximum packet size of a high speed bulk endpoint. I will add a comment.